Branch-SDK/Branch-SDK/BNCStrongMatchHelper.m

```diff
@@ -112,11 +112,23 @@ - (void)presentSafariVCWithBranchKey:(NSString *)branchKey {
         
         // Must be on next run loop to avoid a warning
         dispatch_async(dispatch_get_main_queue(), ^{
-            [windowRootController presentViewController:safController animated:YES completion:^{
-                [self.secondWindow.rootViewController dismissViewControllerAnimated:YES completion:NULL];
-                [BNCPreferenceHelper preferenceHelper].lastStrongMatchDate = [NSDate date];
-                self.requestInProgress = NO;
-            }];
+            // Add the safari view controller using view controller containment
+            [windowRootController addChildViewController:safController];
+            [windowRootController.view addSubview:safController.view];
+            [safController didMoveToParentViewController:windowRootController];
+            
+            // Remove the safari view controller from view controller containment
```

Comment:
Bad news, this isn't long enough. If you wrap the removal in a dispatch after .3 seconds, it will work:
```
            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(.3 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{

                // Remove the safari view controller from view controller containment
                [safController willMoveToParentViewController:nil];
                [safController.view removeFromSuperview];
                [safController removeFromParentViewController];
                
                // Remove the window and release it's strong reference. This is important to ensure that
                // applications using view controller based status bar appearance are restored.
                [self.secondWindow removeFromSuperview];
                self.secondWindow = nil;
                
                [BNCPreferenceHelper preferenceHelper].lastStrongMatchDate = [NSDate date];
                self.requestInProgress = NO;
});
```
I hate hacks like this.